### DISCUSSION TITLE
Upgrade styling tech?

### DISCUSSION THREAD

**Discussion by SimHoZebs** [ipod1g/battleship#6] [Ideas]
There are several options that can improve DX in styling, no matter which way you prefer to style things.

1. TailwindCSS
This approach would eliminate writing CSS files as a whole. We'd be writing them as classes instead. I summarized the benefits of it in this document: https://www.notion.so/simhozebs/Tailwind-Windi-197ec217ee6143c7a4c92c2cb2bd940a

2. SCSS
This would keep our current approach, writing a separate CSS file. You can see this as a superset of CSS, like how Typescript is to JavaScript. The basic features it supports can be read here: https://sass-lang.com/guide

3. Emotion (CSS-in-JS)
This would similar to SCSS, except our styles would be written in JS. I do not want this approach; Even the creator of the library himself is against it: https://dev.to/srmagura/why-were-breaking-up-wiht-css-in-js-4g9b
I just thought it was worth mentioning.

Overall, I'm a fan of TailwindCSS :D

**Comment by ipod1g** [comment]
I am thinking of refurnishing the stylings to TailwindCSS, but only after finishing the game logic perhaps? Hence, I am going to stick to the classic CSS styles at this brief moment of development.
Thank you for offering these nice choices

**Reply by SimHoZebs** [reply]
That works! I was actually thinking of moving to SCSS as well because there's barely any work for it. I could migrate our styles to SCSS first to improve DX with minimal migration work.

Did you get to check it out? What do you think?

**Reply by ipod1g** [reply]
Sure, SCSS doesn't seem too deviated from the general CSS format so it should be fine. Additionally, we don't have much styling right now anyway, so any choice you think improves DX is cool.
